cbt_enable_and_snapshot.py

```diff
@@ -14,6 +14,12 @@
 import requests
 import argparse
 
+def collect_host_certificates(session, certfile="cacert.pem"):

```

Comment:
If we are going to delete the file holding them immediately afterwards doesn't feel like their is any point parametising it. Will hardcode instead.

Reply:
Either way, I think it makes sense to have one variable (or constant) and use it both for writing to and reading from the file, i.e. here in `with open(certfile, 'w') as cert_out:` and also in the line which is currently
`client = new_nbd_client(host, export_name, "./cacert.pem")`

Reply:
Also, this pattern of fetching the host certificates and using them as CA certificates only works with self-signed certificates; otherwise when calling `new_nbd_client` we would need instead to specify the relevant CA certificate(s) obtained elsewhere (and there would be no need to fetch the host certs). I think it might be worth having a comment saying that.

Reply:
If get_server_certificate only works with self-signed certificates, should we rename the call?
Also we need to make sure we guide partners, so they understand that they may need to get the CA from elsewhere.

Reply:
I will rename it to `collect_self_signed_host_certificates` and add a comment with some details. For the TP I don't think we need to worry about explaining how to deal with non-self signed certs, although obviously it will need to be in the final docs.

Reply:
I think that function has a good name: it does indeed fetch the host certificates, regardless of whether they are self-signed.

For `context.load_verify_locations(cafile=self.ca_cert)` we need a file containing the CA certificate(s) that signed the host certificates. If a host certificate is self-signed, this is a special case: the host certificate is the CA certificate that signed the host certificate.